### PR TITLE
Implement SDFC date <-> Go time type

### PR DESCRIPTION
**Pull request by dragonsinth**
See https://github.com/nimajalali/go-force/issues/38

### CODE REVIEW

File: sobjects/base.go [hunk 1]
```diff
@@ -26,9 +26,9 @@ type BaseSObject struct {
 	Id               string            `force:",omitempty" json:",omitempty"`
 	IsDeleted        bool              `force:",omitempty" json:",omitempty"`
 	Name             string            `force:",omitempty" json:",omitempty"`
-	CreatedDate      string            `force:",omitempty" json:",omitempty"`
+	CreatedDate      *Time             `force:",omitempty" json:",omitempty"`
 	CreatedById      string            `force:",omitempty" json:",omitempty"`
-	LastModifiedDate string            `force:",omitempty" json:",omitempty"`
+	LastModifiedDate *Time             `force:",omitempty" json:",omitempty"`
```
Comment (dragonsinth) [inline sobjects/base.go]: These have to be pointers rather than embedded structs for `omitempty` to be respected.

File: sobjects/date.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
+package sobjects
+
+import (
+	"time"
+	"golang.org/x/exp/errors/fmt"
+	"github.com/nimajalali/go-force/forcejson"
+)
+
+const SFTIMEFORMAT1 = "2006-01-02T15:04:05.000-0700"
+const SFTIMEFORMAT2 = "2006-01-02T15:04:05.999Z"
```
Comment (dragonsinth) [inline sobjects/date.go]: https://developer.salesforce.com/docs/atlas.en-us.api_bulk_v2.meta/api_bulk_v2/datafiles_date_format.htm

File: sobjects/date.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+// Represents a SFDC Date. Implements marshaling / unmarshaling as a Go Time.
+type Time time.Time
```
Comment (dragonsinth) [inline sobjects/date.go]: There are pros and cons to this formulation vs. 
```go
type struct Time {
     time.Time
}
```

Feedback welcome.

File: sobjects/date.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	b = append(b, '"')
+	b = time.Time(t).AppendFormat(b, SFTIMEFORMAT1)
+	b = append(b, '"')
+	return b, nil
+}
+
+// Convenience Go time.Time converstion.
+func (t Time) Time() time.Time {
+	return time.Time(t)
+}
```
Comment (dragonsinth) [inline sobjects/date.go]: This should possibly be:

```go
func (t* Time) Time() time.Time {
	if t == nil {
		return time.Time{}
	}
	return time.Time(*t)
}
```

**Comment by dragonsinth** [comment]
@nimajalali any interest in this?

**Comment by dragonsinth** [comment]
Is this project active?

File: sobjects/date.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+	tm, err := ParseTime(str)
+	if err != nil {
+		return err
+	}
+	*t = *tm
+	return nil
+}
+
+var _ forcejson.Unmarshaler = (*Time)(nil)
```
Comment (nimajalali) [inline sobjects/date.go:83]: Are these required? What are they accomplishing?

File: sobjects/date_test.go [hunk 6]
```diff
@@ -0,0 +1,38 @@
+package sobjects
+
+import (
+	"testing"
```
Comment (nimajalali) [inline sobjects/date_test.go:4]: Minor but can we run goimports on this?

File: sobjects/date.go (hunk 5 above)
Comment (dragonsinth) [reply sobjects/date.go:83]: They're just compile time interface checks.  We're verifying that the type implements custom json marshal/unmarshal.

File: sobjects/date_test.go (hunk 6 above)
Comment (dragonsinth) [reply sobjects/date_test.go:4]: Sure!

**Comment by dragonsinth** [comment]
updated and rebased

**Comment by nimajalali** [comment]
@dragonsinth I fixed the test account and moved CI to github actions. Do you mind rebasing with master then we can merge?

**Comment by dragonsinth** [comment]
Re-rebased!

**Comment by nimajalali** [comment]
Merged! Thx

**Comment by dragonsinth** [comment]
Awesome thanks!

**Pull request merged**